doc/relevancy.md

```diff
@@ -0,0 +1,442 @@
+terms are mapped to the same single value that exists in the index.
+
+Comma separated "foo, bar, baz" means to return any of those terms.
+
+### Filtering
+
+Additional configuration is defined in the elasticsearch_schema.yml and
+stems.yml files.  This configuration is merged with the JSON configuration,
+and then passed to elasticsearch directly.
+
+Within elasticsearch_schema.yml, as well as an index_synonym filter there is
+also a synonym_protwords filter.
+
+This marks the terms generated from the synonym matching process as "keywords".
+This prevents them being stemmed, and is required if there is a stemming filter
+following the synonym filter in the analysis chain. This will make sure phrases
+stay together aka "heart attack".
+
+This blog post suggests that as well as using keywords, it can be useful to use
+'keep words' to only filter the phrases we want.
+
+The post suggests to first generate shingles, followed by synonyms, followed
+by keepwords.
+
+In other words: generate candidate keyphrases by shingling, expand them with
+synonyms, then cull out any non-synonyms with keepwords.
+
+### Old synonyms
+
+Synonyms are also specified within old_synonyms.yml which contains a set of
+synonyms which were applied in an old, and moderately broken, way.
+
+According to the comments:
+
+"This is kept separate from the `synonyms.yml` file so that we can support
+and modify the lists for the new synonym approach without changing things
+for the old approach.  Once we're happy with the new approach, we'll delete
+the `old_synonyms.yml` file, and remove support for using it." (Nov 2016)
+
+The analyzer which use the old_synonyms filter do not also use the
+synonym_protwords which could mean less effective results.
+
+Not too sure why we have not migrated the old_synonyms.
+
+### Summary
+
+Things to further investigate:
+
+- Are the synonymn groupings effective. Does simple contraction make much difference?
+- Currently with_index_synonyms filter looks like this: filter: [standard,
+asciifolding, lowercase, index_synonym, synonym_protwords, stemmer_override,
+stemmer_english]. Maybe we could try applying shingles, then synonyms and then
+applying keepwords (like suggested in the blog post)
+- Figure out what is going on with old_synonyms.
+- Is popularity boosted for synonyms of the term?
+
+
+## Popularity boosting
+
+How does it work?
+
+Popularity indicator is used to bias search results. It is one of the factors
+involved in weighting search results. A higher number indicates more pageviews.
+It is updated nightly.
+
+Popularity is derived from recent page views. The gov.uk search uses page
+popularity information extracted from Google. The data is gathered by the
+search-analytics application. That data is imported into Search API.
+
+### Popularity score
+
+`popularity_score` seems to be set in `PopularityLookup` class.
+
+```ruby
+popularity_score = 1.0 / (ranks[link] + @search_config.popularity_rank_offset)
```

Comment:
This is still correct: `popularity_score = 1.0 / (ranks[link] + SearchConfig.popularity_rank_offset)`